lib/ukboot/include/uk/boot/earlytab.h

```diff
@@ -0,0 +1,48 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2024, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#ifndef __UK_BOOT_EARLYTAB_H__
+#define __UK_BOOT_EARLYTAB_H__
+
+#include <uk/config.h>
+#include <uk/plat/common/bootinfo.h>
+#include <uk/plat/common/common.lds.h>
+#include <uk/essentials.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef int (*uk_boot_earlytab_func_t)(struct ukplat_bootinfo *bi);
+
+struct uk_boot_earlytab_entry {
```

Comment:
Would we need the struct?

Reply:
Since we align the struct to 8 bytes I don't think it makes any difference. Below you can see two entries registered from two different libraries:
```
Hex dump of section '.uk_boot_earlytab':
  0x001621b8 10861200 00000000 c0851200 00000000 ................
``` 
It's more of a matter of habit / good practice in case we need to extend later. Saying that, I don't have an extension in mind atm, so if you have a strong preference we can easily switch to functions.

Reply:
This is actually true, since it is a single-field `struct` with a pointer, you should be able to cast it to a function pointer even: no real difference in memory. I am also undecided which version is better, normally I tend to minimalism but here we have zero runtime overhead due to the `struct`. Could be worth keeping it. 🤔 Let's keep it, as is.